compiler/resolution/functionResolution.cpp

```diff
@@ -8913,10 +8919,18 @@ static void insertDynamicDispatchCalls() {
         BlockStmt* ifBlock = new BlockStmt();
         VarSymbol* cid = newTemp("_dynamic_dispatch_tmp_", dtBool);
         ifBlock->insertAtTail(new DefExpr(cid));
-        ifBlock->insertAtTail(new CallExpr(PRIM_MOVE, cid,
-                                new CallExpr(PRIM_TESTCID,
-                                             call->get(2)->copy(),
-                                             type->symbol)));
+
+        Expr* getCid = NULL;
+        if (isSuperAccess) {
+          // We're in a call on the super type.  We already know the answer to
+          // this if conditional
+          getCid = new SymExpr(gFalse);
```

Comment:
For those following along at home, while we wanted to change this block to only insert the false branch, apparently this code does some mystical magic to insert the "right" actions into the true and false branches after the fact.  Fixing that can of worms is beyond the scope of this PR.